packages/miniflare/src/plugins/kv/assets.ts

```diff
@@ -97,3 +98,122 @@ export function getAssetsServices(options: AssetsOptions): Service[] {
 	};
 	return [storageService, namespaceService];
 }
+
+const MAX_ASSET_COUNT = 20_000;
+const MAX_ASSET_SIZE = 25 * 1024 * 1024;
+const MANIFEST_HEADER_SIZE = 20;
+
+const PATH_HASH_OFFSET = 0;
+const PATH_HASH_SIZE = 16;
+
+const CONTENT_HASH_OFFSET = PATH_HASH_SIZE;
+const CONTENT_HASH_SIZE = 16;
+
+const TAIL_RESERVED_SIZE = 8;
+
+const ENTRY_SIZE = PATH_HASH_SIZE + CONTENT_HASH_SIZE + TAIL_RESERVED_SIZE;
+
+type AssetManifestEntry = { contentHash: Uint8Array; pathHash: Uint8Array };
+
+const walk = async (dir: string) => {
+	const files = await fs.readdir(dir, { recursive: true });
+	const manifest: AssetManifestEntry[] = [];
+	let counter = 0;
+	await Promise.all(
+		files.map(async (file) => {
+			const filepath = path.join(dir, file);
+			const relativeFilepath = path.relative(dir, filepath);
+			const filestat = await fs.stat(filepath);
+
+			if (filestat.isSymbolicLink() || filestat.isDirectory()) {
+				return;
+			} else {
+				if (counter >= MAX_ASSET_COUNT) {
```

Comment:
yeah, I thought about this too. In my original attempt to crack the assets manifest, I passed both `MAX_ASSET_COUNT` and `MAX_ASSET_SIZE` as options, alongside `assetsPath`. The thinking there was that we'd have "one" source of truth, and we'd de-risk running out of sync between wrangler and miniflare. But in the light of us working towards making ASW a Miniflare primitive, I think it's OK for these to be internal to Miniflare. We "just" need to remember we have to update these values in two places, if we ever need to update them

Reply:
i'll add a comment to the corresponding wrangler bit for posterity.